frontend/packages/dev-console/src/components/import/devfile/devfileHooks.ts

```diff
@@ -21,24 +24,85 @@ export const useDevfileServer = (
 
   const {
     name,
-    git: { url, ref, dir },
+    git: { url, ref, dir, type, secretResource },
     devfile,
   } = values;
   const smartSlashDir = prefixDotSlash(suffixSlash(dir));
 
   const { devfileContent, devfilePath } = devfile || {};
 
-  const devfileData = React.useMemo(() => {
+  const getResourceContent = React.useCallback(
+    async (resourceURI: string) => {
+      let resourceContent;
+      if (resourceURI.startsWith('http://') || resourceURI.startsWith('https://')) {
+        try {
+          resourceContent = await coFetch(resourceURI);
+        } catch (e) {
+          // eslint-disable-next-line no-console
+          console.error(e);
+        }
+      } else {
+        const gitService = getGitService(url, type, ref, dir, secretResource, devfilePath);
+        let resourcePath;
+        if (resourceURI.startsWith('/')) {
+          resourcePath = resourceURI;
+        } else {
+          const contextDir = dir.replace(/\/$/, '');
+          resourcePath = `${contextDir}/${resourceURI}`;
+        }
+        resourceContent = gitService && (await gitService.getFileContent(resourcePath));
+      }
+      return resourceContent;
+    },
+    [devfilePath, dir, ref, secretResource, type, url],
+  );
+
+  const getParsedComponents = React.useCallback(
+    async (component) => {
+      let resourceContent;
+      if (component.kubernetes && component.kubernetes.uri) {
+        resourceContent = await getResourceContent(component.kubernetes.uri);
+        if (resourceContent) {
+          component.kubernetes = _.omit({ ...component.kubernetes, inlined: resourceContent }, [
+            'uri',
+          ]);
+        }
+      } else if (component.openshift && component.openshift.uri) {
+        resourceContent = await getResourceContent(component.openshift.uri);
+        if (resourceContent) {
+          component.openshift = _.omit({ ...component.openshift, inlined: resourceContent }, [
+            'uri',
+          ]);
+        }
+      }
+      return component;
+    },
+    [getResourceContent],
+  );
+
+  const convertURItoInlineYAML = React.useCallback(async () => {
+    const devfileJSON = safeLoad(devfileContent);
+    for (let component of devfileJSON.components) {
+      // eslint-disable-next-line no-await-in-loop
+      component = await getParsedComponents(component);
+    }
+
+    return safeDump(devfileJSON);
```

Comment:
Sorry but I didnt understand the ask here. Could you please elaborate what should be changed/updated?

Reply:
You can remove `component =` and it will still work. Because `getParsedComponents` (btw. it should be singular) modifies the argument, so returning it and assigning it here doesn't make sense. I would expect that a "getter" returns a new object instead.

It's great to mark everything which is _really_ resolved as resolved, but IMHO it make finding such discussion easier if you keep them open until we agree this doesn't need a change.

But in this case, it's okay for me to keep it.